libraries-2/src/test/java/com/baeldung/okhttp/ResponseDecoderUnitTest.java

```diff
@@ -0,0 +1,160 @@
+package com.baeldung.okhttp;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.InputStreamReader;
+
+public class ResponseDecoderUnitTest {
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Rule
+    public MockWebServer server = new MockWebServer();
+
+    SimpleEntity sampleResponse;
+
+    MockResponse mockResponse;
+
+    OkHttpClient client;
+
+
+    @Before
+    public void setUp() {
+        sampleResponse = new SimpleEntity("Baeldung");
+        client = new OkHttpClient.Builder()
+                .build();
```

Comment:
indentation on continuation lines should be only two spaces, so let's fix this throughout all the Java files